pkg/catalog/catalog.go

```diff
@@ -1868,14 +1869,17 @@ func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID string,
 		return nil, err
 	}
 
-	uncommittedLocation, err := c.Store.GCGetUncommittedLocation(repository, runID)
-	if err != nil {
-		return nil, err
-	}
-
 	// Upload parquet file to object store
-	var name string
+	var (
+		uncommittedLocation string
+		name                string
+	)
 	if hasData {
+		uncommittedLocation, err = c.Store.GCGetUncommittedLocation(repository, runID)
```

Comment:
This means that no location == no data, right?

Reply:
True